### ISSUE TITLE
Not building

### ISSUE THREAD

**Issue by ryanpcmcquen** [p12tic/libdesktop-agnostic#4]
I thought it was a vala issue, but I have tried with vala 0.28.0 and vala 0.12.1

vala 0.12.1:
```
[ 17/115] valac: tests/test-vfs-trash.vala -> build/default/tests/test-vfs-trash.c
../libdesktop-agnostic/config-bridge.vala:182.34-182.50: warning: deprecated syntax, use [CCode (has_target = false)]
../tests/test-vfs-trash.vala:39.7-39.51: warning: deprecated syntax, use `connect' method instead
../libdesktop-agnostic/config-schema-option.vala:146.34-146.50: warning: string.offset is deprecated. Use string.substring
../libdesktop-agnostic/config-schema.vala:233.41-233.50: warning: string.rchr is deprecated. Use string.last_index_of_char
../libdesktop-agnostic/config-schema.vala:234.27-234.49: warning: string.pointer_to_offset is deprecated. Use string.char_count
../libdesktop-agnostic/config-schema.vala:236.41-236.52: warning: string.offset is deprecated. Use string.substring
../libdesktop-agnostic/config-bridge.vala:250.9-250.69: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/config-client.vala:128.21-128.45: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/config-client.vala:329.9-329.74: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/config-client.vala:337.9-337.78: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/config-client.vala:372.9-372.65: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/config-client.vala:380.9-380.69: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/config-client.vala:76.31-76.65: warning: unhandled error `GLib.Error'
Waf: Leaving directory `/tmp/SBo/libdesktop-agnostic-0.3.94/build'
Build failed
 -> task failed (err #1): 
        {task: valac_task vfs-impl-gio.vala,vfs-file-impl-gio.vala,vfs-file-monitor-impl-gio.vala,vfs-trash-impl-gio.vala,vfs-volume-impl-gio.vala -> vfs-impl-gio.c,da-vfs-gio.h,da-vfs-gio.vapi,da-vfs-gio.deps,vfs-file-impl-gio.c,da-vfs-gio.h,da-vfs-gio.vapi,da-vfs-gio.deps,vfs-file-monitor-impl-gio.c,da-vfs-gio.h,da-vfs-gio.vapi,da-vfs-gio.deps,vfs-trash-impl-gio.c,da-vfs-gio.h,da-vfs-gio.vapi,da-vfs-gio.deps,vfs-volume-impl-gio.c,da-vfs-gio.h,da-vfs-gio.vapi,da-vfs-gio.deps,da-vfs-gio.h}
```

vala 0.28.0:
```
[ 25/115] valac: tests/test-color.vala -> build/default/tests/test-color.c
[ 26/115] valac: libdesktop-agnostic/config-type-color.vala -> build/default/libdesktop-agnostic/config-type-color.c build/default/libdesktop-agnostic/da-cfg-type-c
olor.h build/default/libdesktop-agnostic/da-cfg-type-color.vapi build/default/libdesktop-agnostic/da-cfg-type-color.deps
../libdesktop-agnostic/desktop-entry-impl-glib.vala:192.14-192.47: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:286.15-286.54: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:296.32-296.64: warning: unhandled error `GLib.ShellError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:302.15-302.50: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:304.26-304.64: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:305.29-305.54: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:336.44-336.48: warning: string.len is deprecated. Use string.length
../libdesktop-agnostic/desktop-entry-impl-glib.vala:460.42-460.46: warning: string.len is deprecated. Use string.length
../libdesktop-agnostic/desktop-entry-impl-glib.vala:451.12-451.48: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:59.15-59.63: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:60.15-61.75: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:65.15-65.81: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:65.15-65.81: warning: unhandled error `GLib.FileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:93.13-94.73: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/desktop-entry-impl-glib.vala:113.50-113.59: warning: string.size is deprecated. Use string.length
../libdesktop-agnostic/desktop-entry-impl-glib.vala:113.13-114.73: warning: unhandled error `GLib.KeyFileError'
../tests/test-color.vala:41.9-41.50: warning: unhandled error `GLib.Error'
../libdesktop-agnostic/config-type-color.vala:81.32-81.69: warning: unhandled error `GLib.KeyFileError'
../libdesktop-agnostic/config-type-color.vala:111.18-111.41: warning: unhandled error `DesktopAgnostic.Config.SchemaError'
../libdesktop-agnostic/config-type-color.vala:120.18-120.51: warning: unhandled error `DesktopAgnostic.Config.SchemaError'
Waf: Leaving directory `/tmp/SBo/libdesktop-agnostic-0.3.94/build'
Build failed
 -> task failed (err #1): 
        {task: valac_task ui-color-button.vala,ui-icon-button.vala,ui-icon-chooser-dialog.vala,ui-launcher-editor-dialog.vala -> ui-color-button.c,ui.h,desktop-agno
stic-ui.vapi,desktop-agnostic-ui.deps,ui-icon-button.c,ui.h,desktop-agnostic-ui.vapi,desktop-agnostic-ui.deps,ui-icon-chooser-dialog.c,ui.h,desktop-agnostic-ui.vapi
,desktop-agnostic-ui.deps,ui-launcher-editor-dialog.c,ui.h,desktop-agnostic-ui.vapi,desktop-agnostic-ui.deps,ui.h}
```

**Comment by kozec** [comment]
So, those are all warnings, but treated as errors? That should be fixable by changing one cflag...

**Comment by ryanpcmcquen** [comment]
Builds fine with vala 0.26.2. What CFLAG would you suggest? This is the
script:
http://slackbuilds.org/slackbuilds/14.1/libraries/libdesktop-agnostic/libdesktop-agnostic.SlackBuild

**Comment by kozec** [comment]
Is that entire build log? I'm getting errors little before those warnings...

```
[ 24/121] valac: libdesktop-agnostic/desktop-entry-impl-glib.vala -> build/default/libdesktop-agnostic/desktop-entry-impl-glib.c build/default/libdesktop-agnostic/da-fdo-glib.h build/default/libdesktop-agnostic/da-fdo-glib.vapi build/default/libdesktop-agnostic/da-fdo-glib.deps
[ 25/121] valac: tests/test-desktop-entry.vala -> build/default/tests/test-desktop-entry.c
../libdesktop-agnostic/ui-icon-chooser-dialog.vala:168.43-168.51: error: `ListStore' is an ambiguous reference between `GLib.ListStore' and `Gtk.ListStore'
../libdesktop-agnostic/ui-icon-chooser-dialog.vala:208.13-208.21: error: `ListStore' is an ambiguous reference between `GLib.ListStore' and `Gtk.ListStore'
../libdesktop-agnostic/ui-icon-chooser-dialog.vala:294.15-294.23: error: `ListStore' is an ambiguous reference between `GLib.ListStore' and `Gtk.ListStore'
../libdesktop-agnostic/ui-icon-chooser-dialog.vala:299.42-299.50: error: `ListStore' is an ambiguous reference between `GLib.ListStore' and `Gtk.ListStore'
../libdesktop-agnostic/ui-icon-chooser-dialog.vala:345.15-345.23: error: `ListStore' is an ambiguous reference between `GLib.ListStore' and `Gtk.ListStore'
../libdesktop-agnostic/ui-icon-chooser-dialog.vala:349.44-349.52: error: `ListStore' is an ambiguous reference between `GLib.ListStore' and `Gtk.ListStore'
```

**Comment by kozec** [comment]
Fix: #5

**Comment by ryanpcmcquen** [comment]
Thanks @kozec! I can send the whole build log your way if needed.
